### PR TITLE
feat(core): decimal fingerprint for i18n

### PR DESCRIPTION
**Pull request by vicb**
/ref https://github.com/angular/angular/issues/12511

@mprobst @gnoack

Do you think there is any interest in having the same fingerprint as in g3 ? (the algo is the same but placeholders are not serialized in the same way and meaning is also "mixed" in a different way)

### CODE REVIEW

**Comment by vicb** [comment]
Should be ok with decimal ids now but I want to fix #12512 in this PR as well to avoid changing the IDs mid-flight

**Comment by mprobst** [comment]
I think it'd be nice to have the fingerprint be compatible with g3, as that'd safe a bunch of re-translating of messages already translated in some other context. Depends on how hard that is to achieve, I guess?

File: modules/@angular/compiler/test/i18n/extractor_merger_spec.ts [hunk 1]
```diff
@@ -93,9 +92,10 @@ export function main() {
               ],
               [
                 [
-                  'text',
-                  '<ph tag name="START_PARAGRAPH">html, <ph tag name="START_BOLD_TEXT">nested</ph name="CLOSE_BOLD_TEXT"></ph name="CLOSE_PARAGRAPH">',
-                  '<ph icu name="ICU">{count, plural, =0 {[<ph tag name="START_TAG_SPAN">html</ph name="CLOSE_TAG_SPAN">]}}</ph>',
+                  'text', '<ph tag name="START_PARAGRAPH">html, <ph tag' +
+                      ' name="START_BOLD_TEXT">nested</ph name="CLOSE_BOLD_TEXT"></ph name="CLOSE_PARAGRAPH">',
```
Comment (panuruj) [inline modules/@angular/compiler/test/i18n/extractor_merger_spec.ts]: The closing tag looks invalid.
`</ph name="CLOSE_BOLD_TEXT">`

File: modules/@angular/compiler/test/i18n/digest_spec.ts [hunk 2]
```diff
@@ -8,51 +8,89 @@
...
+         () => { expect(sha1('')).toEqual('da39a3ee5e6b4b0d3255bfef95601890afd80709'); });
+
+      it('should returns the sha1 of "hello world"',
+         () => { expect(sha1('abc')).toEqual('a9993e364706816aba3e25717850c26c9cd0d89d'); });
+
+      it('should returns the sha1 of unicode strings',
+         () => { expect(sha1('你好，世界')).toEqual('3becb03b015ed48050611c8d7afe4b88f70d5a20'); });
+
+      it('should support arbitrary string size', () => {
+        // node.js reference code:
```
Comment (vicb) [inline modules/@angular/compiler/test/i18n/digest_spec.ts]: run on node

File: modules/@angular/compiler/test/i18n/extractor_merger_spec.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/compiler/test/i18n/extractor_merger_spec.ts]: At this point, this is not supposed to be valid HTML, only a serialization of the AST

File: modules/@angular/compiler/test/ml_parser/ast_serializer_spec.ts [hunk 3]
```diff
@@ -97,4 +97,4 @@ const serializerVisitor = new _SerializerVisitor();
 
 export function serializeNodes(nodes: html.Node[]): string[] {
   return nodes.map(node => node.visit(serializerVisitor, null));
-}
+}
```
Comment (alfaproject) [inline modules/@angular/compiler/test/ml_parser/ast_serializer_spec.ts]: This is what I was trying to avoid. (;

File: modules/@angular/platform-browser/src/dom/events/hammer_common.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 import {EventManagerPlugin} from './event_manager';
 
-var _eventNames = {
+const _eventNames = {
```
Comment (alfaproject) [inline modules/@angular/platform-browser/src/dom/events/hammer_common.ts]: Maybe do this in another PR?

**Comment by alfaproject** [comment]
@vicb can I start adding `gettext` support based on this?

**Comment by vicb** [comment]
@alfaproject yep, this should get merged later this week after the release
